include/alpaka/mem/view/Accessor.hpp

```diff
@@ -0,0 +1,186 @@
+    //! returned on access.
+    //! @tparam TBufferIdx The integral type used for indexing and index computations.
+    //! @tparam TDim The dimensionality of the accessed data.
+    //! @tparam TAccessModes Either a single access tag type or a `std::tuple` containing multiple access tag types.
+    template<typename TMemoryHandle, typename TElem, typename TBufferIdx, std::size_t TDim, typename TAccessModes>
+    struct Accessor;
+
+    namespace traits
+    {
+        //! The customization point for how to build an accessor for a given memory object.
+        template<typename TMemoryObject, typename SFINAE = void>
+        struct BuildAccessor
+        {
+            template<typename... TAccessModes, typename TMemoryObjectForwardRef>
+            ALPAKA_FN_HOST_ACC static auto buildAccessor(TMemoryObjectForwardRef&&)
+            {
+                static_assert(
+                    meta::DependentFalseType<TMemoryObject>::value,
+                    "BuildAccessor<TMemoryObject> is not specialized for your TMemoryObject.");
+            }
+        };
+    } // namespace traits
+
+    namespace internal
+    {
+        template<typename AccessorOrBuffer>
+        struct MemoryHandle
+        {
+        };
+
+        template<typename TMemoryHandle, typename TElem, typename TBufferIdx, std::size_t TDim, typename TAccessModes>
+        struct MemoryHandle<Accessor<TMemoryHandle, TElem, TBufferIdx, TDim, TAccessModes>>
+        {
+            using type = TMemoryHandle;
+        };
+    } // namespace internal
+
+    /// Get the memory handle type of the given accessor or buffer type.
+    template<typename Accessor>
+    using MemoryHandle = typename internal::MemoryHandle<Accessor>::type;
+
+    namespace internal
+    {
+        template<typename T>
+        struct IsAccessor : std::false_type
+        {
+        };
+
+        template<typename TMemoryHandle, typename TElem, typename TBufferIdx, std::size_t Dim, typename TAccessModes>
+        struct IsAccessor<Accessor<TMemoryHandle, TElem, TBufferIdx, Dim, TAccessModes>> : std::true_type
+        {
+        };
+    } // namespace internal
+
+    //! Creates an accessor for the given memory object using the specified access modes. Memory objects are e.g.
+    //! alpaka views and buffers.
+    template<
+        typename... TAccessModes,
+        typename TMemoryObject,
+        typename = std::enable_if_t<!internal::IsAccessor<std::decay_t<TMemoryObject>>::value>>
+    ALPAKA_FN_HOST_ACC auto accessWith(TMemoryObject&& memoryObject)
+    {
+        return traits::BuildAccessor<std::decay_t<TMemoryObject>>::template buildAccessor<TAccessModes...>(
+            memoryObject);
+    }
+
+    namespace internal
+    {
+        template<typename List, typename Value>
+        struct MpContains : std::false_type
+        {
+        };
+
+        template<template<typename...> class List, typename Head, typename... Tail, typename Value>
+        struct MpContains<List<Head, Tail...>, Value>
```

Comment:
This would also be a pefect fit for a new header within the `meta` folder, a new unit test and could be added to the `Front` PR.